### ISSUE TITLE
Release ROS Lunar MoveIt!

### ISSUE THREAD

**Issue by davetcoleman** [ros-planning/moveit#486]
- [x] RViz https://github.com/ros-visualization/rviz/issues/1090
- [x] octomap_msgs https://github.com/OctoMap/octomap_msgs/issues/8
- [x] object_recognition_msgs https://github.com/wg-perception/object_recognition_msgs/issues/5
- [x] warehouse_ros https://github.com/ros-planning/warehouse_ros/issues/36
- [x] moveit_msgs @130s --> https://github.com/ros/rosdistro/pull/14529
- [x] moveit_resources @130s --> https://github.com/ros/rosdistro/pull/14532
- ompl 
  - [x] https://bitbucket.org/ompl/ompl/issues/318/release-into-ros-lunar --> https://github.com/ros/rosdistro/pull/14958
  - [x] Build issue on armhf https://github.com/ros-gbp/ompl-release/issues/7
- [x] srdfdom @130s --> https://github.com/ros/rosdistro/pull/14530
- FCL
  - [x] https://github.com/ros-planning/moveit/issues/486#issuecomment-294607434
  - [x] Re-run bloom with new rosdep keys https://github.com/ros-planning/moveit/issues/486#issuecomment-300853362 --> Done https://github.com/ros-planning/moveit/issues/486#issuecomment-301181911
- [x] <s>household_objects_database_msgs https://github.com/ros-interactive-manipulation/household_objects_database_msgs/issues/5</s> --> no need
- [x] <s>manipulation_msgs https://github.com/ros-interactive-manipulation/manipulation_msgs/issues/10</s> --> no need
- [x] python-assimp for Ubuntu YZ https://github.com/ros/rosdistro/pull/14963
- [x] libglew-dev --> https://github.com/ros/rosdistro/pull/14963
- [x] [visualization on Ubuntu YK] compilation issue (`boost`?) https://github.com/ros-planning/moveit/issues/486#issuecomment-300911832, binary released https://github.com/ros-planning/moveit/issues/517#issuecomment-306334668
- [x] MSA on Ubuntu YK, Debian S #506 

Post-release:
- [x] Website install instructions -> delegated to https://github.com/ros-planning/moveit.ros.org/pull/127 
----

It looks like we are approaching MoveIt!'s turn to release into Lunar, [geometric_shapes](https://github.com/ros-planning/geometric_shapes/issues/65) is now a blocker for URDFDom... @130s do you have cycles to release that soon?

As discussed in our last maintainer meeting, our plan is to release lunar out of the current ``kinetic-devel`` branch for as long as possible.

Subject for discussion: as soon as we release MoveIt! into Lunar, I'd support reducing @130s' effort by no longer releasing updates into the Jade branch, including cherry-picks

**Comment by 130s** [comment]
+1 for the kindness @davetcoleman :), and for dropping Jade support soon.
Also +1 for being kind for all MoveIt! users as much as possible. I/we can make a release into Lunar soon, hopefully before its official release (I assume that asking around maintainers of all reverse-dependent packages to make releases might take some time).

**Comment by 130s** [comment]
Asking around some direct dependencies for the release into Lunar. I'm sure there will be more once I look into non-direct dependency:
- RViz https://github.com/ros-visualization/rviz/issues/1090
- octomap_msgs https://github.com/OctoMap/octomap_msgs/issues/8
- object_recognition_msgs https://github.com/wg-perception/object_recognition_msgs/issues/5
- warehouse_ros https://github.com/ros-planning/warehouse_ros/issues/36
- moveit_msgs @130s
- moveit_resources @130s
- ompl https://bitbucket.org/ompl/ompl/issues/318/release-into-ros-lunar
- srdfdom @130s

**Comment by mikaelarguedas** [comment]
> +1 for the kindness @davetcoleman :), and for dropping Jade support soon.

Jade will be EOL at the same time as Lunar is released and packages won't be built for it as of end of May.

> Also +1 for being kind for all MoveIt! users as much as possible. I/we can make a release into Lunar soon, hopefully before its official release

If this is possible, that would be really awesome, thanks for pounding on that!

**Comment by 130s** [comment]
- [x] FCL, which is used to be a ROS package but now standalone, so might be worth being noted.
  - Looks like for all Ubuntu X (0.3.2), [Y](http://packages.ubuntu.com/source/yakkety/fcl), [Z](http://packages.ubuntu.com/source/zesty/fcl) (0.5.0 for both) that [Lunar supports](http://www.ros.org/reps/rep-0003.html#lunar-loggerhead-may-2017-may-2019) it's already released.
  - I can't remember if Debian is now required (REP says "recommendation" but I remember `bloom` wants a released package on Debian)

**Comment by mikaelarguedas** [comment]
Debian is a "recommended support", but we try to get as many packages as possible released into debian.

Is there any blocker to get Moveit! working on debian stretch ? looks like libfcl is the same as in Y and Z (0.5.0) on [debian stretch](https://packages.debian.org/stretch/libfcl-dev) so updating the rosdep key should be all you need.

**Comment by 130s** [comment]
> looks like libfcl is the same as in Y and Z (0.5.0) on [debian stretch](https://packages.debian.org/stretch/libfcl-dev) so updating the rosdep key should be all you need.

@mikaelarguedas Thanks for clarification. Looks like the necessary `fcl` is already available and [registered on rosdistro](https://github.com/ros/rosdistro/blob/89acff9ab83bbf336a25b7559ebf74ac6a49d82c/rosdep/base.yaml#L1457).

> Is there any blocker to get Moveit! working on debian stretch ? 

I'm not sure since I've been holding from running `bloom` for Lunar knowing that all the dependency hasn't been cleared yet.

**Comment by 130s** [comment]
- [ ] household_objects_database_msgs https://github.com/ros-interactive-manipulation/household_objects_database_msgs/issues/5

**Comment by 130s** [comment]
- [ ] manipulation_msgs https://github.com/ros-interactive-manipulation/manipulation_msgs/issues/10

**Comment by 130s** [comment]
- [ ] python-assimp for Ubuntu YZ https://github.com/ros/rosdistro/pull/14963

**Comment by 130s** [comment]
Running prerelease tests locally:
- Lunar-Xenial passed ([full log](https://drive.google.com/open?id=0Bw_hbVS3wTLydmtDbl8xWlN5Q2s).
- Lunar-Zesty failed for missing rosdep keys ([full log](https://drive.google.com/open?id=0Bw_hbVS3wTLybWJpcUx2ZExCWlk))

**Comment by 130s** [comment]
- [ ] libglew-dev --> https://github.com/ros/rosdistro/pull/14963

**Comment by 130s** [comment]
Done https://github.com/ros/rosdistro/pull/14966

**Comment by mikaelarguedas** [comment]
FYI: you'll need to rerun bloom once https://github.com/ros/rosdistro/pull/14967 has been merged, for moveit to compile on Yakkety, Zesty and Debian Stretch

**Comment by 130s** [comment]
> FYI: you'll need to rerun bloom once ros/rosdistro#14967 has been merged, for moveit to compile on Yakkety, Zesty and Debian Stretch

Thanks for the heads-up @mikaelarguedas. I'm curious if there was any issue? If so where can I see it indicated?

I admit that after https://github.com/ros-planning/moveit/issues/486#issuecomment-300648476 where I once ran prerelease test against `Zesty`, I resolved rosdep keys but then forgot to re-run prerelease test but instead I just proceeded and confirmed that `bloom` was able to resolve all `rosdep` keys. Clearly this doesn't mean that I confirmed MoveIt! building on Z (as well as Y).

Also I thought I/we confirmed about FCL situation in a previous post https://github.com/ros-planning/moveit/issues/486#issuecomment-294607434, so if that wasn't sufficient I want to know what else needed to be done back then.

**Comment by mikaelarguedas** [comment]
As far as I understand, fcl 0.5.x was required for kinetic. At that time wily and xenial were providing 0.3.x so we had to package libfcl 0.5 ourselves for these Ubuntu distributions http://repositories.ros.org/ubuntu/main/pool/main/f/fcl/. Now that fcl 0.5 is upstream for yakkety, zesty and stretch, we will use the upstream version. To do so we need to update the rosdep key to map to libfcl-dev for these distributions.
> Also I thought I/we confirmed about FCL situation in a previous post #486 (comment), so if that wasn't sufficient I want to know what else needed to be done back then.

The only thing we could have done was to update the rosdep keys as suggested in https://github.com/ros-planning/moveit/issues/486#issuecomment-294629339

> I admit that after #486 (comment) where I once ran prerelease test against Zesty, I resolved rosdep keys but then forgot to re-run prerelease test but instead I just proceeded and confirmed that bloom was able to resolve all rosdep keys. Clearly this doesn't mean that I confirmed MoveIt! building on Z (as well as Y).

Bloom passed because the rosdep entry "exist" in the rosdep database but the jobs fail because the package it refers to doesn't exist. What you did was adding the rosdep keys that did not "exist" in the rosdep database but it didn't mean that the ones that existed were accurate.

> I'm curious if there was any issue? If so where can I see it indicated?

Several jobs failed after the release because of the unability to install libfcl-0.5-dev on these platforms. 
A good way to get notified when a package fails is to be listed as a maintainer of the package. Apparently the recipients of the failure notification email were:
* dave@dav.ee,
* me@v4hn.de,
* mferguson@fetchrobotics.com

Another way (less convenient) is to look at the status pages: e.g. http://repositories.ros.org/status_page/ros_lunar_default.html?q=moveit_core. This shows you that the job passed only on xenial but failed on yakkety and zesty. By clicking on the red box under Y64 you are redirected to the failing job.

Hope this helps

**Comment by mikaelarguedas** [comment]
@130s FYI: https://github.com/ros/rosdistro/pull/14967 has been merged

**Comment by 130s** [comment]
- ompl 
  - [x] https://bitbucket.org/ompl/ompl/issues/318/release-into-ros-lunar --> https://github.com/ros/rosdistro/pull/14958
  - [ ] Build issue on armhf https://github.com/ros-gbp/ompl-release/issues/7

**Comment by 130s** [comment]
> FYI: ros/rosdistro#14967 has been merged

@mikaelarguedas Does it make sense to `bloom` without OMPL issue https://github.com/ros-gbp/ompl-release/issues/7 not resolved yet? If so I can certainly re-bloom.

**Comment by mikaelarguedas** [comment]
I think it does given that armhf users is less that 3% of the user base. Having moveit working for the remaining 97% is sensible IMO.

**Comment by 130s** [comment]
Ok, thanks for the cheerup @mikaelarguedas :)

Hm, I wanted to be cautious so ran a prerelease test against `Zesty`, then it failed ([full log](https://drive.google.com/open?id=0Bw_hbVS3wTLyNFo0d29FMWM2ZWM)). I'm just re-running it locally now.

```
:
==> cmake /tmp/catkin_workspace/src/moveit/moveit_ros/visualization -DCATKIN_DEVEL_PREFIX=/tmp/catkin_workspace/devel_isolated/moveit_ros_visualization -DCMAKE_INSTALL_PREFIX=/tmp/catkin_workspace/install_isolated -DCATKIN_SKIP_TESTING=1 -G Unix Makefiles in '/tmp/catkin_workspace/build_isolated/moveit_ros_visualization'
:
==> make -j1 in '/tmp/catkin_workspace/build_isolated/moveit_ros_visualization'
Scanning dependencies of target moveit_rviz_plugin_render_tools_automoc
[  2%] Automatic moc for target moveit_rviz_plugin_render_tools
Generating moc source moveit_rviz_plugin_render_tools_automoc.dir/moc_trajectory_vis_NE6OZUMGTDEUX5.cpp
Generating moc compilation moveit_rviz_plugin_render_tools_automoc.cpp
[  2%] Built target moveit_rviz_plugin_render_tools_automoc
Scanning dependencies of target moveit_rviz_plugin_render_tools
[  4%] Building CXX object rviz_plugin_render_tools/CMakeFiles/moveit_rviz_plugin_render_tools.dir/src/render_shapes.cpp.o
In file included from /usr/include/boost/config.hpp:61:0,
                 from /usr/include/boost/detail/workaround.hpp:41,
                 from /usr/include/boost/range/config.hpp:14,
                 from /usr/include/boost/range/begin.hpp:18,
                 from /usr/include/boost/range/functions.hpp:18,
                 from /usr/include/boost/range.hpp:18,
                 from /usr/include/OGRE/OgreIteratorRange.h:48,
                 from /usr/include/OGRE/OgreIteratorWrappers.h:33,
                 from /usr/include/OGRE/OgreRenderQueue.h:32,
                 from /usr/include/OGRE/OgreMovableObject.h:34,
                 from /usr/include/OGRE/OgreSimpleRenderable.h:33,
                 from /opt/ros/lunar/include/rviz/ogre_helpers/point_cloud.h:33,
                 from /tmp/catkin_workspace/src/moveit/moveit_ros/visualization/rviz_plugin_render_tools/include/moveit/rviz_plugin_render_tools/octomap_render.h:42,
                 from /tmp/catkin_workspace/src/moveit/moveit_ros/visualization/rviz_plugin_render_tools/include/moveit/rviz_plugin_render_tools/render_shapes.h:40,
                 from /tmp/catkin_workspace/src/moveit/moveit_ros/visualization/rviz_plugin_render_tools/src/render_shapes.cpp:37:
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_half<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:265:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(half, 5.000000000000000000000000000000000000e-01, "5.00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:265:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_third<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:266:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(third, 3.333333333333333333333333333333333333e-01, "3.33333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333333e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:266:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_twothirds<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:267:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(twothirds, 6.666666666666666666666666666666666666e-01, "6.66666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666667e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:267:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_two_thirds<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:268:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(two_thirds, 6.666666666666666666666666666666666666e-01, "6.66666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666667e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:268:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_three_quarters<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:269:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(three_quarters, 7.500000000000000000000000000000000000e-01, "7.50000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:269:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_root_two<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:270:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(root_two, 1.414213562373095048801688724209698078e+00, "1.41421356237309504880168872420969807856967187537694807317667973799073247846210703885038753432764157273501384623e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:270:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_root_three<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:271:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(root_three, 1.732050807568877293527446341505872366e+00, "1.73205080756887729352744634150587236694280525381038062805580697945193301690880003708114618675724857567562614142e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:271:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_half_root_two<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:272:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(half_root_two, 7.071067811865475244008443621048490392e-01, "7.07106781186547524400844362104849039284835937688474036588339868995366239231053519425193767163820786367506923115e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:272:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_ln_two<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:273:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(ln_two, 6.931471805599453094172321214581765680e-01, "6.93147180559945309417232121458176568075500134360255254120680009493393621969694715605863326996418687542001481021e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:273:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_ln_ln_two<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:274:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(ln_ln_two, -3.665129205816643270124391582326694694e-01, "-3.66512920581664327012439158232669469454263447837105263053677713670561615319352738549455822856698908358302523045e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:274:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_root_ln_four<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:275:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(root_ln_four, 1.177410022515474691011569326459699637e+00, "1.17741002251547469101156932645969963774738568938582053852252575650002658854698492680841813836877081106747157858e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:275:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_one_div_root_two<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:276:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(one_div_root_two, 7.071067811865475244008443621048490392e-01, "7.07106781186547524400844362104849039284835937688474036588339868995366239231053519425193767163820786367506923115e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:276:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:277:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(pi, 3.141592653589793238462643383279502884e+00, "3.14159265358979323846264338327950288419716939937510582097494459230781640628620899862803482534211706798214808651e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:277:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_half_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:278:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(half_pi, 1.570796326794896619231321691639751442e+00, "1.57079632679489661923132169163975144209858469968755291048747229615390820314310449931401741267105853399107404326e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:278:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_third_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:279:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(third_pi, 1.047197551196597746154214461093167628e+00, "1.04719755119659774615421446109316762806572313312503527365831486410260546876206966620934494178070568932738269550e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:279:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_sixth_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:280:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(sixth_pi, 5.235987755982988730771072305465838140e-01, "5.23598775598298873077107230546583814032861566562517636829157432051302734381034833104672470890352844663691347752e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:280:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_two_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:281:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(two_pi, 6.283185307179586476925286766559005768e+00, "6.28318530717958647692528676655900576839433879875021164194988918461563281257241799725606965068423413596429617303e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:281:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_two_thirds_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:282:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(two_thirds_pi, 2.094395102393195492308428922186335256e+00, "2.09439510239319549230842892218633525613144626625007054731662972820521093752413933241868988356141137865476539101e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:282:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_three_quarters_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:283:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(three_quarters_pi, 2.356194490192344928846982537459627163e+00, "2.35619449019234492884698253745962716314787704953132936573120844423086230471465674897102611900658780098661106488e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:283:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_four_thirds_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:284:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(four_thirds_pi, 4.188790204786390984616857844372670512e+00, "4.18879020478639098461685784437267051226289253250014109463325945641042187504827866483737976712282275730953078202e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:284:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_one_div_two_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:285:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(one_div_two_pi, 1.591549430918953357688837633725143620e-01, "1.59154943091895335768883763372514362034459645740456448747667344058896797634226535090113802766253085956072842727e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:285:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_one_div_root_two_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:286:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(one_div_root_two_pi, 3.989422804014326779399460599343818684e-01, "3.98942280401432677939946059934381868475858631164934657665925829670657925899301838501252333907306936430302558863e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:286:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_root_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:287:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(root_pi, 1.772453850905516027298167483341145182e+00, "1.77245385090551602729816748334114518279754945612238712821380778985291128459103218137495065673854466541622682362e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:287:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_root_half_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:288:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(root_half_pi, 1.253314137315500251207882642405522626e+00, "1.25331413731550025120788264240552262650349337030496915831496178817114682730392098747329791918902863305800498633e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:288:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_root_two_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:289:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(root_two_pi, 2.506628274631000502415765284811045253e+00, "2.50662827463100050241576528481104525300698674060993831662992357634229365460784197494659583837805726611600997267e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:289:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_log_root_two_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:290:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(log_root_two_pi, 9.189385332046727417803297364056176398e-01, "9.18938533204672741780329736405617639861397473637783412817151540482765695927260397694743298635954197622005646625e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:290:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_one_div_root_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:291:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(one_div_root_pi, 5.641895835477562869480794515607725858e-01, "5.64189583547756286948079451560772585844050629328998856844085721710642468441493414486743660202107363443028347906e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:291:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_root_one_div_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:292:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(root_one_div_pi, 5.641895835477562869480794515607725858e-01, "5.64189583547756286948079451560772585844050629328998856844085721710642468441493414486743660202107363443028347906e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:292:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_pi_minus_three<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:293:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(pi_minus_three, 1.415926535897932384626433832795028841e-01, "1.41592653589793238462643383279502884197169399375105820974944592307816406286208998628034825342117067982148086513e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:293:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_four_minus_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:294:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(four_minus_pi, 8.584073464102067615373566167204971158e-01, "8.58407346410206761537356616720497115802830600624894179025055407692183593713791001371965174657882932017851913487e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:294:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_pi_pow_e<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:296:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(pi_pow_e, 2.245915771836104547342715220454373502e+01, "2.24591577183610454734271522045437350275893151339966922492030025540669260403991179123185197527271430315314500731e+01")
   ^
/usr/include/boost/math/constants/constants.hpp:296:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_pi_sqr<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:297:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(pi_sqr, 9.869604401089358618834490999876151135e+00, "9.86960440108935861883449099987615113531369940724079062641334937622004482241920524300177340371855223182402591377e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:297:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_pi_sqr_div_six<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:298:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(pi_sqr_div_six, 1.644934066848226436472415166646025189e+00, "1.64493406684822643647241516664602518921894990120679843773555822937000747040320087383362890061975870530400431896e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:298:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_pi_cubed<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:299:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(pi_cubed, 3.100627668029982017547631506710139520e+01, "3.10062766802998201754763150671013952022252885658851076941445381038063949174657060375667010326028861930301219616e+01")
   ^
/usr/include/boost/math/constants/constants.hpp:299:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_cbrt_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:300:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(cbrt_pi, 1.464591887561523263020142527263790391e+00, "1.46459188756152326302014252726379039173859685562793717435725593713839364979828626614568206782035382089750397002e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:300:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_one_div_cbrt_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:301:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(one_div_cbrt_pi, 6.827840632552956814670208331581645981e-01, "6.82784063255295681467020833158164598108367515632448804042681583118899226433403918237673501922595519865685577274e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:301:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_e<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:302:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(e, 2.718281828459045235360287471352662497e+00, "2.71828182845904523536028747135266249775724709369995957496696762772407663035354759457138217852516642742746639193e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:302:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_exp_minus_half<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:303:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(exp_minus_half, 6.065306597126334236037995349911804534e-01, "6.06530659712633423603799534991180453441918135487186955682892158735056519413748423998647611507989456026423789794e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:303:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_e_pow_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:304:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(e_pow_pi, 2.314069263277926900572908636794854738e+01, "2.31406926327792690057290863679485473802661062426002119934450464095243423506904527835169719970675492196759527048e+01")
   ^
/usr/include/boost/math/constants/constants.hpp:304:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_root_e<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:305:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(root_e, 1.648721270700128146848650787814163571e+00, "1.64872127070012814684865078781416357165377610071014801157507931164066102119421560863277652005636664300286663776e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:305:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_log10_e<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:306:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(log10_e, 4.342944819032518276511289189166050822e-01, "4.34294481903251827651128918916605082294397005803666566114453783165864649208870774729224949338431748318706106745e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:306:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_one_div_log10_e<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:307:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(one_div_log10_e, 2.302585092994045684017991454684364207e+00, "2.30258509299404568401799145468436420760110148862877297603332790096757260967735248023599720508959829834196778404e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:307:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_ln_ten<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:308:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(ln_ten, 2.302585092994045684017991454684364207e+00, "2.30258509299404568401799145468436420760110148862877297603332790096757260967735248023599720508959829834196778404e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:308:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_degree<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:309:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(degree, 1.745329251994329576923690768488612713e-02, "1.74532925199432957692369076848861271344287188854172545609719144017100911460344944368224156963450948221230449251e-02")
   ^
/usr/include/boost/math/constants/constants.hpp:309:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_radian<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:310:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(radian, 5.729577951308232087679815481410517033e+01, "5.72957795130823208767981548141051703324054724665643215491602438612028471483215526324409689958511109441862233816e+01")
   ^
/usr/include/boost/math/constants/constants.hpp:310:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_sin_one<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:311:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(sin_one, 8.414709848078965066525023216302989996e-01, "8.41470984807896506652502321630298999622563060798371065672751709991910404391239668948639743543052695854349037908e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:311:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_cos_one<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:312:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(cos_one, 5.403023058681397174009366074429766037e-01, "5.40302305868139717400936607442976603732310420617922227670097255381100394774471764517951856087183089343571731160e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:312:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_sinh_one<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:313:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(sinh_one, 1.175201193643801456882381850595600815e+00, "1.17520119364380145688238185059560081515571798133409587022956541301330756730432389560711745208962339184041953333e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:313:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_cosh_one<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:314:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(cosh_one, 1.543080634815243778477905620757061682e+00, "1.54308063481524377847790562075706168260152911236586370473740221471076906304922369896426472643554303558704685860e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:314:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_phi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:315:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(phi, 1.618033988749894848204586834365638117e+00, "1.61803398874989484820458683436563811772030917980576286213544862270526046281890244970720720418939113748475408808e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:315:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_ln_phi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:316:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(ln_phi, 4.812118250596034474977589134243684231e-01, "4.81211825059603447497758913424368423135184334385660519661018168840163867608221774412009429122723474997231839958e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:316:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_one_div_ln_phi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:317:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(one_div_ln_phi, 2.078086921235027537601322606117795767e+00, "2.07808692123502753760132260611779576774219226778328348027813992191974386928553540901445615414453604821933918634e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:317:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_euler<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:318:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(euler, 5.772156649015328606065120900824024310e-01, "5.77215664901532860606512090082402431042159335939923598805767234884867726777664670936947063291746749514631447250e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:318:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_one_div_euler<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:319:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(one_div_euler, 1.732454714600633473583025315860829681e+00, "1.73245471460063347358302531586082968115577655226680502204843613287065531408655243008832840219409928068072365714e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:319:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_euler_sqr<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:320:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(euler_sqr, 3.331779238077186743183761363552442266e-01, "3.33177923807718674318376136355244226659417140249629743150833338002265793695756669661263268631715977303039565603e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:320:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_zeta_two<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:321:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(zeta_two, 1.644934066848226436472415166646025189e+00, "1.64493406684822643647241516664602518921894990120679843773555822937000747040320087383362890061975870530400431896e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:321:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_zeta_three<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:322:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(zeta_three, 1.202056903159594285399738161511449990e+00, "1.20205690315959428539973816151144999076498629234049888179227155534183820578631309018645587360933525814619915780e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:322:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_catalan<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:323:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(catalan, 9.159655941772190150546035149323841107e-01, "9.15965594177219015054603514932384110774149374281672134266498119621763019776254769479356512926115106248574422619e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:323:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_glaisher<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:324:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(glaisher, 1.282427129100622636875342568869791727e+00, "1.28242712910062263687534256886979172776768892732500119206374002174040630885882646112973649195820237439420646120e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:324:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_khinchin<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:325:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(khinchin, 2.685452001065306445309714835481795693e+00, "2.68545200106530644530971483548179569382038229399446295305115234555721885953715200280114117493184769799515346591e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:325:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_extreme_value_skewness<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:326:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(extreme_value_skewness, 1.139547099404648657492793019389846112e+00, "1.13954709940464865749279301938984611208759979583655182472165571008524800770607068570718754688693851501894272049e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:326:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_rayleigh_skewness<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:327:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(rayleigh_skewness, 6.311106578189371381918993515442277798e-01, "6.31110657818937138191899351544227779844042203134719497658094585692926819617473725459905027032537306794400047264e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:327:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_rayleigh_kurtosis<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:328:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(rayleigh_kurtosis, 3.245089300687638062848660410619754415e+00, "3.24508930068763806284866041061975441541706673178920936177133764493367904540874159051490619368679348977426462633e+00")
   ^
/usr/include/boost/math/constants/constants.hpp:328:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_rayleigh_kurtosis_excess<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:329:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(rayleigh_kurtosis_excess, 2.450893006876380628486604106197544154e-01, "2.45089300687638062848660410619754415417066731789209361771337644933679045408741590514906193686793489774264626328e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:329:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_two_div_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:331:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(two_div_pi, 6.366197723675813430755350534900574481e-01, "6.36619772367581343075535053490057448137838582961825794990669376235587190536906140360455211065012343824291370907e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:331:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
/usr/include/boost/math/constants/constants.hpp: In static member function ‘static constexpr T boost::math::constants::detail::constant_root_two_div_pi<T>::get(const mpl_::int_<5>&)’:
/usr/include/boost/math/constants/constants.hpp:332:3: error: unable to find numeric literal operator ‘operator""Q’
   BOOST_DEFINE_MATH_CONSTANT(root_two_div_pi, 7.978845608028653558798921198687637369e-01, "7.97884560802865355879892119868763736951717262329869315331851659341315851798603677002504667814613872860605117725e-01")
   ^
/usr/include/boost/math/constants/constants.hpp:332:3: note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes
make[2]: *** [rviz_plugin_render_tools/CMakeFiles/moveit_rviz_plugin_render_tools.dir/src/render_shapes.cpp.o] Error 1
rviz_plugin_render_tools/CMakeFiles/moveit_rviz_plugin_render_tools.dir/build.make:62: recipe for target 'rviz_plugin_render_tools/CMakeFiles/moveit_rviz_plugin_render_tools.dir/src/render_shapes.cpp.o' failed
make[1]: *** [rviz_plugin_render_tools/CMakeFiles/moveit_rviz_plugin_render_tools.dir/all] Error 2
CMakeFiles/Makefile2:3145: recipe for target 'rviz_plugin_render_tools/CMakeFiles/moveit_rviz_plugin_render_tools.dir/all' failed
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
<== Failed to process package 'moveit_ros_visualization':
  Command '['/tmp/catkin_workspace/install_isolated/env.sh', 'make', '-j1']' returned non-zero exit status 2

Reproduce this error by running:
==> cd /tmp/catkin_workspace/build_isolated/moveit_ros_visualization && /tmp/catkin_workspace/install_isolated/env.sh make -j1

Command failed, exiting.
# END SUBSECTION
```

**Comment by mikaelarguedas** [comment]
Looks like a boost issue to me. Maybe something changed in boost 1.62 that's why the error shows up only on Zesty. I guess you already tried using the suggested flags (`note: use -std=gnu++11 or -fext-numeric-literals to enable more built-in suffixes`)?

**Comment by 130s** [comment]
Hm, I don't have much gears anytime soon to try fixing compilation issues. 

- If someone can take a look, that'll be appreciated.
- Elif no one can, then might as well remove rosdistro entry for now for Lunar until it gets fixed.

Btw just memo (to myself); on buildfarm (not prerelease test) I see the [builds on Y and Z are failing](http://repositories.ros.org/status_page/ros_lunar_default.html?q=moveit) for Lunar (e.g. [moveit_core on May 11](http://build.ros.org/view/Lbin_uZ64/job/Lbin_uZ64__moveit_core__ubuntu_zesty_amd64__binary/2/consoleFull)) but this looks like another issue that was mentioned in https://github.com/ros-planning/moveit/issues/486#issuecomment-300811843 and can be fixed by re-`bloom`ing.

**Comment by 130s** [comment]
> Hm, I don't have much gears anytime soon to try fixing compilation issues.
>
> -    If someone can take a look, that'll be appreciated.

To raise attention I opened a separate issue https://github.com/ros-planning/moveit/issues/504

**Comment by 130s** [comment]
Re-`bloom`ed https://github.com/ros/rosdistro/pull/14979, hopefully this fixes one of the issues (regarding FCL https://github.com/ros-planning/moveit/issues/486#issuecomment-300853362).

**Comment by v4hn** [comment]
Weird, I didn't hit this problem before, although I use even more current versions of gcc and boost on my system...

It is probably enough to replace [this line](https://github.com/ros-planning/moveit/blob/kinetic-devel/moveit_ros/visualization/CMakeLists.txt#L4) by `add_compile_options(-std=gnu++11)`, and it is ok to do this here I suppose.
I don't think there are any other packages, but if something else transitively includes the ogre-headers, they will need this too.

Because I can't reproduce the error, I can't test the fix though.

**Comment by mikaelarguedas** [comment]
There is a gist [here](https://gist.github.com/mikaelarguedas/7731b4100e7ed78b6aab73f052ffac79) with the prerelease console outputs for Yakkety and Zesty for anyone wanting to investigate the failures. I haven't ran it for Stratch because I expect it to fail exactly the same way as Zesty.
Errors can be reproduced by running the prerelease locally (following instructions of [prerelease.ros.org/lunar](http://prerelease.ros.org/lunar))

**Comment by v4hn** [comment]
From the prerelease page:

> you can run in the terminal on a Ubuntu machine

Sorry I'm out, I run Lunar-Linux and don't have access to/time to access a Ubuntu machine in the next couple of days.

I wanted to point out that this could actually be a problem of Ubuntu or Debians packages though.
I run gcc 6.3, boost 1.63 and 1.64 and ogre 1.9.0 over here (pretty much zasty's versions) and don't see the error.

Changing the compile flag should work though.

**Comment by mikaelarguedas** [comment]
everything runs inside docker so the platform of your host machine should not matter much as long as it supports docker. I'm testing the fix locally as we speak, I'll update here with the outcome.

**Comment by de-vri-es** [comment]
I'm on vacation right now so I only skimmed the issue, but it may be enough
to compile with `-DBOOST_MATH_DISABLE_FLOAT128` (this is from memory). If
it works I prefer that over enabling gcc specific extensions, since that
hinders compatibility with non GNU platforms.

On 13 May 2017 12:38 a.m., "Mikael Arguedas" <notifications@github.com>
wrote:

everything runs inside docker so the platform of your host machine should
not matter much as long as it supports docker. I'm testing the fix locally
as we speak, I'll update here with the outcome.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/ros-planning/moveit/issues/486#issuecomment-301192047>,
or mute the thread

**Comment by 130s** [comment]
I thought MSA build issue on Ubuntu YZ #506 would block the Lunar release, but for the platforms where binaries were successfully built, it seems we can install, like Ubuntu Xenial.
So, the task in the title "release into Lunar" is basically achieved and we only have a "regression" #506.
I close this ticket, but if that's not desirable don't hesitate to shoot me.
Thanks all who paid attention to this especially @mikaelarguedas

**Comment by mikaelarguedas** [comment]
Thanks @130s for following up on this.
I don't think that this ticket should be closed because, IMHO, releasing into a new distro means having binaries building on all platforms for which there is required support (that are [Ubuntu X Y and Z](http://www.ros.org/reps/rep-0003.html#lunar-loggerhead-may-2017-may-2019) on amd64 architecture).


@davetcoleman @de-vri-es @v4hn and moveit maintainers, is there any plan on your end to build on the missing platforms / support the deboostified version of urdfdom? If yes, do you have an idea of where this could fit in your roadmap and how external contributors can help?


---
@130s 
>  it seems we can install, like Ubuntu Xenial.

We usually don't sync packages that don't build on all platforms, I had to sync the day of the official Lunar release despit the fact that moveit was the only repo not compiling at that point, but I think it is still a blocker for the "Lunar release of moveit". I'm not sure what the policy is in the scenario where this still doesnt compile for the next syncs. The possible outcomes are disabling the moveit builds for the other platforms or completely remove moveit from Lunar until it compiles. Maybe @tfoote or @dirk-thomas  can give us more information in that matter.

**Comment by dirk-thomas** [comment]
> The possible outcomes are disabling the moveit builds for the other platforms

Since the three Ubuntu distros Xenial, Yakkety, and Zesty are being managed in the same [build file](https://github.com/ros-infrastructure/ros_buildfarm_config/blob/8c60b5f914827a469c88cd273568f8c16a5194b9/lunar/release-build.yaml#L52-L59) they can't be disabled separately. I think we can only highly recommend that the `moveit` teams makes an effort to support all of these distros. If they don't the farm will continue to send notifications for the failing jobs. I don't think there is anything we should do about it.

**Comment by v4hn** [comment]
On Fri, May 26, 2017 at 10:14:28PM -0700, Mikael Arguedas wrote:
> is there any plan on your end to build on the missing platforms

Yes. The latest branch should compile on all (reasonable) platforms requested by users.
This clearly includes more current Ubuntu releases and Debian Stretch.

> / support the deboostified version of urdfdom?
> If yes, do you have an idea of where this could fit in your roadmap and how external contributors can help?

I haven't looked into the issue yet, so I can't say anything about the means required.

The standard contributor-guideline "point out the problem and propose a fix" seems to apply here.

**Comment by davetcoleman** [comment]
I will address the MSA bug now, sorry for the delay

**Comment by 130s** [comment]
Looks like all issues are now resolved so finally ready to make full-release into Lunar.

For kinetic-devel, there has only 4 commits since the last release, so making a patch release into Lunar (and Kinetic too?) soon might be good. Or do we want to wait for more commits to be available? Thoughts welcomed.

**Comment by BryceStevenWilley** [comment]
I just ran into an issue with compiling [moveit_planners_ompl](https://github.com/ros-planning/moveit/tree/kinetic-devel/moveit_planners/ompl) with OMPL, both from source. 

Specifically, the [constraints library](https://github.com/ros-planning/moveit/blob/kinetic-devel/moveit_planners/ompl/ompl_interface/src/constraints_library.cpp#L308) tries to make a SpaceStorage instance with a const ref, but as of [OMPL 1.3.0](https://github.com/ompl/ompl/commit/51940fb95ba7d3c6b79b75275ed1205a0a96abd4), most of OMPL was changed to pass by value. Not sure why this hasn't been caught by Jenkins yet. My setup is just a catkin wrapper around OMPL and moveit in a single workspace, and the from-source-lib of OMPL is being passed to moveit_planners_ompl. 

This is something that should be addressed on the OMPL side (we should be returning values, not const ref anyway), but it'd require another OMPL release, which would delay Lunar release even more.

**Comment by v4hn** [comment]
@BryceStevenWilley Could you open a new issue for this please?

**Comment by davetcoleman** [comment]
Since we haven't released a full Lunar release yet, I think we should do that asap now that MSA is fixed

**Comment by davetcoleman** [comment]
All checklist items have been completed, I'm going to declare Lunar released and any additional tasks can be added to new issues. Thanks @130s for all your efforts!
